### PR TITLE
Consolidate payloads

### PR DESCRIPTION
**Pull request by r-ash**
Started adding this to #447 but became a bit of a bigger refactor so broken this out into a separate PR

Atm we have 3 copies of payloads
* Payload in `tests/testthat/testdata` for use in tests
* Payload in `example/*` for use when api started with `api()`
* Payload in `example/docker_*` for use when api started within docker e.g. when generating the README

Ideally these wouldn't be copied about, as it ends up leading to us forgetting to update a payload in one place. So this PR will
* Add 1 version of each payload into `inst` dir
* Add helper functions for building the payload when the payload needs the version number
* Use `test_path` so when running tests interactively we don't have to change the working dir to `tests/testthat` anymore
* Make the README generation work again

### CODE REVIEW

**Comment by r-ash** [comment]
Let's actually do this by setting the root dir when we start the API. On production would be "/", in docker "/src"

File: tests/testthat/integration-server.R [hunk 1]
```diff
@@ -1,6 +1,8 @@
-server <- porcelain::porcelain_background$new(
-  api, args = list(queue_id = paste0("hintr:", ids::random_id())))
-server$start()
+withr::with_dir(testthat::test_path(), {
```
Comment (richfitz) [inline tests/testthat/integration-server.R:1]: we should add something to porcelain_bg to make this nicer

**Review by richfitz — COMMENTED** [review]
As advertised, this is a bit gross yes, but looks mostly mechanical at least :)

Can you check on why that test is unexpectedly failing, and there's a couple of tiny notes here otherwise let's get this in

File: R/payload_helpers.R [hunk 2]
```diff
@@ -0,0 +1,89 @@
+# nocov start
```
[the remark this answers is not in the document]
Comment (r-ash) [reply R/payload_helpers.R:1]: They're called by `scripts/build_readme` script here https://github.com/mrc-ide/hintr/pull/448/files#diff-8fb2d313e5f8ce1b3091f398594ff25d410c8102e9ee7f96eb17213d8bccdf47R46 which runs the `data` section from the `README.yml` e.g. https://github.com/mrc-ide/hintr/pull/448/files#diff-51acf556c815f48bd4b30e2af19e4ab450dcbc9c710e2535704c8944a8dc3fa1R71 so needs to be a function which is part of the package

File: docker/build [hunk 3]
```diff
@@ -24,7 +24,6 @@ RUN install_packages --repo=https://mrc-ide.github.io/drat \
         covr \
         ssh
 WORKDIR /src
-RUN R CMD build .
```
[the remark this answers is not in the document]
Comment (r-ash) [reply docker/build:27]: Not sure, probably just included by mistake

File: tests/testthat/integration-server.R (hunk 1 above)
Comment (r-ash) [reply tests/testthat/integration-server.R:1]: Made a ticket https://mrc-ide.myjetbrains.com/youtrack/issue/RESIDE-333/Add-ability-to-start-porcelain-server-with-specified-working-directory

**Review by richfitz — APPROVED** [review]

**Pull request merged**